> =================================== FAILURES ===================================
_________________ test_account_role_policy_allow_create_bucket _________________

sorry, there are some new suite changes for s3control that will fail until you rebase :(

--- a/PendingReleaseNotes
+++ b/PendingReleaseNotes
@@ -16,6 +16,7 @@
 * RGW: Omap backing for the RGW Datalog is deprecated and support will be removed in a future version.
   - The `rgw default data log backing` option is removed and it is no longer
     possible to create clusters with an omap based datalog.
+  - `radosgw-admin datalog type` will only accept `--log_type=fifo`.
 * RGW: Bucket Logging suppports creating log buckets in EC pools.
   Implicit logging object commits are now performed asynchronously.
 * RGW: radosgw-admin bucket list now supports pagination for versioned buckets by using

--- a/PendingReleaseNotes
+++ b/PendingReleaseNotes
@@ -13,6 +13,9 @@
 * ceph-volume: Raw BlueStore OSD preparation now pre-formats NVMe devices and
   skips the slower BlueStore discard phase,reducing mkfs time on
   very large namespaces.
+* RGW: Omap backing for the RGW Datalog is deprecated and support will be removed in a future version.
+  - The `rgw default data log backing` option is removed and it is no longer
+    possible to create clusters with an omap based datalog.
 * RGW: Bucket Logging suppports creating log buckets in EC pools.
   Implicit logging object commits are now performed asynchronously.
 * RGW: radosgw-admin bucket list now supports pagination for versioned buckets by using

--- a/src/common/options/rgw.yaml.in
+++ b/src/common/options/rgw.yaml.in
@@ -3937,19 +3937,6 @@ options:
   see_also:
   - rgw_dmclock_metadata_res
   - rgw_dmclock_metadata_wgt
-- name: rgw_default_data_log_backing
-  type: str
-  level: advanced
-  desc: Default backing store for the RGW data sync log
-  long_desc: Whether to use the older OMAP backing store or the high performance FIFO
-    based backing store by default. This only covers the creation of the log on startup
-    if none exists.
-  default: fifo
-  services:
-  - rgw
-  enum_values:
-  - fifo
-  - omap
 - name: rgw_d3n_l1_local_datacache_enabled
   type: bool
   level: advanced

--- a/src/rgw/driver/rados/rgw_datalog.cc
+++ b/src/rgw/driver/rados/rgw_datalog.cc
@@ -494,10 +494,6 @@ RGWDataChangesLog::start(const DoutPrefixProvider *dpp,
   down_flag = false;
   ran_background = (recovery || watch || renew);
 
-  auto defbacking = to_log_type(
-    cct->_conf.get_val<std::string>("rgw_default_data_log_backing"));
-  // Should be guaranteed by `set_enum_allowed`
-  ceph_assert(defbacking);
   try {
     loc = co_await rgw::init_iocontext(dpp, *rados, log_pool,
 				       rgw::create, asio::use_awaitable);
@@ -513,7 +509,7 @@ RGWDataChangesLog::start(const DoutPrefixProvider *dpp,
       dpp, *rados, metadata_log_oid(), loc,
       [this](uint64_t gen_id, int shard) {
 	return get_oid(gen_id, shard);
-      }, num_shards, *defbacking, *this);
+      }, num_shards, log_type::fifo, *this);
   } catch (const std::exception& e) {
     ldpp_dout(dpp, -1) << __PRETTY_FUNCTION__
 		       << ": Error initializing backends: " << e.what()

--- a/src/rgw/radosgw-admin/radosgw-admin.cc
+++ b/src/rgw/radosgw-admin/radosgw-admin.cc
@@ -311,7 +311,7 @@ void usage()
   cout << "  datalog list                     list data log\n";
   cout << "  datalog trim                     trim data log\n";
   cout << "  datalog status                   read data log status\n";
-  cout << "  datalog type                     change datalog type to --log_type={fifo,omap}\n";
+  cout << "  datalog type                     change datalog type to --log_type=fifo\n";
   cout << "  datalog semaphore list           List recovery semaphores\n";
   cout << "  datalog semaphore reset          Reset recovery semaphore (use marker)\n";
   cout << "  orphans find                     deprecated -- init and run search for leaked rados objects (use job-id, pool)\n";
@@ -11427,10 +11427,14 @@ next:
       std::cerr << "log-type not specified." << std::endl;
       return -EINVAL;
     }
+    if (opt_log_type == log_type::omap) {
+      std::cerr << "omap datalogs are deprecated. You cannot convert to them." << std::endl;
+      return -EINVAL;
+    }
     auto datalog = static_cast<rgw::sal::RadosStore*>(driver)->svc()->datalog_rados;
     std::string errstr;
     ret = run_coro(dpp(), context_pool,
-		   datalog->change_format(dpp(), *opt_log_type),
+		   datalog->change_format(dpp(), log_type::fifo),
 		   &errstr);
     if (ret < 0) {
       cerr << "ERROR: change_format(): " << errstr << std::endl;

--- a/src/test/cli/radosgw-admin/help.t
+++ b/src/test/cli/radosgw-admin/help.t
@@ -170,7 +170,7 @@
     datalog list                     list data log
     datalog trim                     trim data log
     datalog status                   read data log status
-    datalog type                     change datalog type to --log_type={fifo,omap}
+    datalog type                     change datalog type to --log_type=fifo
     datalog semaphore list           List recovery semaphores
     datalog semaphore reset          Reset recovery semaphore (use marker)
     orphans find                     deprecated -- init and run search for leaked rados objects (use job-id, pool)

